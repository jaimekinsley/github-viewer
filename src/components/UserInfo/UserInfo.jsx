import React from 'react';
import PropTypes from 'prop-types';

const UserInfo = ({ username, followers, following, link }) => (
  <>
    <p>Name: {username}</p>
    <p>Number of Followers: {followers}</p>
    <p>Number of Following: {following}</p>
    <p>GitHub Profile: {link}</p>
  </>
);

UserInfo.propTypes = {
  username: PropTypes.string.isRequired,
  followers: PropTypes.number.isRequired,
  following: PropTypes.number.isRequired,
  link: PropTypes.string.isRequired
};

export default UserInfo;
