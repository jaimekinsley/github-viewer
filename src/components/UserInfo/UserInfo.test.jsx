import React from 'react';
import { shallow } from 'enzyme';
import UserInfo from './UserInfo';

describe('ShortForm component', () => {
  it('renders UserInfo', () => {
    const wrapper = shallow(<UserInfo username="jaimekinsley" followers={3} following={2} link="http://github.com/jaimekinsley" />);
    expect(wrapper).toMatchSnapshot();
  });
});
