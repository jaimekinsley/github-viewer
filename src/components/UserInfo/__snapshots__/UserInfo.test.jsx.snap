// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ShortForm component renders UserInfo 1`] = `
<Fragment>
  <p>
    Name: 
    jaimekinsley
  </p>
  <p>
    Number of Followers: 
    3
  </p>
  <p>
    Number of Following: 
    2
  </p>
  <p>
    GitHub Profile: 
    http://github.com/jaimekinsley
  </p>
</Fragment>
`;
